File: Csharp/sigmf/sigmf/ext/global/core/HardwareSpec.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    
    [Serializable()]
    public class HardwareSpec
    {
        [JsonProperty("id", NullValueHandling = NullValueHandling.Ignore)]
        protected string id;

        [JsonProperty("model", NullValueHandling = NullValueHandling.Ignore)]
        protected string model;

        [JsonProperty("version", NullValueHandling = NullValueHandling.Ignore)]
        protected string version;

        [JsonProperty("description", NullValueHandling = NullValueHandling.Ignore)]
        protected string description;

        [JsonProperty("supplemental_information", NullValueHandling = NullValueHandling.Ignore)]
        protected string supplemental_information;

        public string Id { get => id; set => id = value; }
        public string Model { get => model; set => model = value; }
        public string Version { get => version; set => version = value; }
        public string Description { get => description; set => description = value; }
        public string Supplemental_information { get => supplemental_information; set => supplemental_information = value; }
    }
}

File: java/src/main/java/gov/doc/ntia/sigmf/ext/core/HardwareSpec.java
package gov.doc.ntia.sigmf.ext.core;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import javax.validation.constraints.NotNull;
import java.io.Serializable;



@JsonInclude(JsonInclude.Include.NON_NULL)
public class HardwareSpec implements Serializable {

  private static final long serialVersionUID = 1L;

  @NotNull
  @JsonProperty(value = "id", required = true)
  protected String id;

  @JsonProperty(value = "model", required = false)
  protected String model;

  @JsonProperty(value = "version", required = false)
  protected String version;

  @JsonProperty(value = "description", required = false)
  protected String description;

  @JsonProperty(value = "supplemental_information", required = false)
  protected String supplementalInformation;

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public String getModel() {
    return model;
  }

  public void setModel(String model) {
    this.model = model;
  }

  public String getVersion() {
    return version;
  }

  public void setVersion(String version) {
    this.version = version;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getSupplementalInformation() {
    return supplementalInformation;
  }

  public void setSupplementalInformation(String supplementalInformation) {
    this.supplementalInformation = supplementalInformation;
  }
}
